main.go

```diff
@@ -173,7 +173,9 @@ func main() {
 	keycloakProfileService := login.NewKeycloakUserProfileClient()
 
 	// Try to fetch the initial list of clusters and start Cluster Service cache refresher
-	err = clusterservice.Start(context.Background(), config)
+	err = clusterservice.Start(context.Background(), config, func(c *http.Client) {
+		c.Timeout = 3 * time.Second
```

Comment:
so is this the timeout that is going to be used for all fetches or only the first one ?

Reply:
so, if I understand correctly, the `auth` service may start with an empty list of clusters in case the `cluster` service was not available at the startup time. Then, if the list is empty when a user request is processed, then `auth` would explicitly call the `cluster` service to get the list of clusters, and return an error if no data was available ? My point is: maybe we could check that the list of cluster is loaded before marking the `auth` service as `ready` (in the `readiness` probes), but I'm not sure it's such a good idea, because there are other parts of the service (eg: roles/permissions) that would still work (i.e., `auth` would be only partially unavailable). WDYT?

Reply:
I hate Lenova touchpad! It's impossible to type without accidentally touching it and messing things up... OK. I need to type again my long response to this comment with disabled touchpad now :(

> so is this the timeout that is going to be used for all fetches or only the first one ?

It's for the first one only.

> so, if I understand correctly, the auth service may start with an empty list of clusters in case the cluster service was not available at the startup time. Then, if the list is empty when a user request is processed, then auth would explicitly call the cluster service to get the list of clusters, and return an error if no data was available ?

There are three places where we try to load the clusters.
- In main() during service startup. It can fail if the cluster service is not available for some reason. Not critical. We can leave it as is (the empty list) and start the service.
- Cluster cache. We pull a new list of clusters every few minutes in a separate goroutine in case there is any changes. It can fail as well if the cluster service is temporally unavailable. In this case we keep using the old cached list of clusters and will try to pull it again in a few minutes.
- When we really need the list of clusters (when handling a user request for example). In this case we use the cached list but if it's empty then we load it from the cluster service. If it fails then we return an error to the user.

>My point is: maybe we could check that the list of cluster is loaded before marking the auth service as ready (in the readiness probes), but I'm not sure it's such a good idea, because there are other parts of the service (eg: roles/permissions) that would still work (i.e., auth would be only partially unavailable). WDYT?

We can't relay on cluster service availability when starting Auth. It will bring a cycle dependency. Cluster service needs Auth service during startup to load the keys. But if Auth needs Cluster service to startup then you can start them both up. They will wait for each other to pass the redness prob.